.travis.yml

```diff
@@ -111,75 +134,77 @@ install:
+  - PKGDIR_lens="$(find . -maxdepth 1 -type d -regex '.*/lens-[0-9.]*')"
+  - PKGDIR_lens_examples="$(find . -maxdepth 1 -type d -regex '.*/lens-examples-[0-9.]*')"
+  - PKGDIR_lens_properties="$(find . -maxdepth 1 -type d -regex '.*/lens-properties-[0-9.]*')"
   # Generate cabal.project
   - rm -rf cabal.project cabal.project.local cabal.project.freeze
   - touch cabal.project
   - |
-    echo 'packages: "lens-*/*.cabal"' >> cabal.project
-    echo 'packages: "lens-examples-*/*.cabal"' >> cabal.project
-    echo 'packages: "lens-properties-*/*.cabal"' >> cabal.project
+    echo "packages: ${PKGDIR_lens}" >> cabal.project
+    echo "packages: ${PKGDIR_lens_examples}" >> cabal.project
+    echo "packages: ${PKGDIR_lens_properties}" >> cabal.project
   - |
-    echo "write-ghc-environment-files: always" >> cabal.project
-  - "for pkg in $($HCPKG list --simple-output); do echo $pkg | sed 's/-[^-]*$//' | grep -vE -- '^(lens|lens-examples|lens-properties)$' | sed 's/^/constraints: /' | sed 's/$/ installed/' >> cabal.project.local; done"
+  - "for pkg in $($HCPKG list --simple-output); do echo $pkg | sed 's/-[^-]*$//' | (grep -vE -- '^(lens|lens-examples|lens-properties)$' || true) | sed 's/^/constraints: /' | sed 's/$/ installed/' >> cabal.project.local; done"
   - cat cabal.project || true
   - cat cabal.project.local || true
   # Building with tests and benchmarks...
   # build & run tests, build benchmarks
-  - ${CABAL} v2-build -w ${HC} ${TEST} ${BENCH} all | color_cabal_output
+  - ${CABAL} v2-build $WITHCOMPILER ${TEST} ${BENCH} all | color_cabal_output
   # Testing...
-  - ${CABAL} v2-test -w ${HC} ${TEST} ${BENCH} all | color_cabal_output
+  - if ! $GHCJS && [ $HCNUMVER -ge 70400 ] ; then ${CABAL} v2-test $WITHCOMPILER ${TEST} ${BENCH} all | color_cabal_output ; fi
   # HLint..
-  - if [ $HCNUMVER -eq 80801 ] ; then (cd lens-* && hlint src) ; fi
-  - if [ $HCNUMVER -eq 80801 ] ; then (cd lens-examples-* && hlint .) ; fi
-  - if [ $HCNUMVER -eq 80801 ] ; then (cd lens-properties-* && hlint src) ; fi
+  - if ! $GHCJS && [ $HCNUMVER -ge 80800 ] && [ $HCNUMVER -lt 80802 ] ; then (cd ${PKGDIR_lens} && hlint src) ; fi
+  - if ! $GHCJS && [ $HCNUMVER -ge 80800 ] && [ $HCNUMVER -lt 80802 ] ; then (cd ${PKGDIR_lens_examples} && hlint .) ; fi
+  - if ! $GHCJS && [ $HCNUMVER -ge 80800 ] && [ $HCNUMVER -lt 80802 ] ; then (cd ${PKGDIR_lens_properties} && hlint src) ; fi
   # cabal check...
-  - (cd lens-* && ${CABAL} -vnormal check)
-  - (cd lens-examples-* && ${CABAL} -vnormal check)
-  - (cd lens-properties-* && ${CABAL} -vnormal check)
+  - (cd ${PKGDIR_lens} && ${CABAL} -vnormal check)
+  - (cd ${PKGDIR_lens_examples} && ${CABAL} -vnormal check)
+  - (cd ${PKGDIR_lens_properties} && ${CABAL} -vnormal check)
   # haddock...
-  - ${CABAL} v2-haddock -w ${HC} ${TEST} ${BENCH} all | color_cabal_output
+  - if ! $GHCJS && [ $HCNUMVER -ge 70400 ] ; then ${CABAL} v2-haddock $WITHCOMPILER ${TEST} ${BENCH} all | color_cabal_output ; fi
```

Comment:
Isn't this `$HCNUMVER -ge 70400` check redundant, given that the minimum version of GHC that `lens` supports is 7.4?

Reply:
I tried to make generation of these checks nicer, i’ll Check why this isn’t as pretty as it could be